src/Core/Label.js

```diff
@@ -0,0 +1,146 @@
+
+const coord = new Coordinates('EPSG:4326');
+let rect;
+
+// set it once
+let STYLE_TRANSFORM = '';
+if (document.documentElement.style.transform !== undefined) {
+    STYLE_TRANSFORM = 'transform';
+} else if (document.documentElement.style.webkitTransform !== undefined) {
+    STYLE_TRANSFORM = 'webkitTransform';
+} else if (document.documentElement.style.mozTransform !== undefined) {
+    STYLE_TRANSFORM = 'mozTransform';
+} else if (document.documentElement.style.oTransform !== undefined) {
+    STYLE_TRANSFORM = 'oTransform';
+} else {
+    STYLE_TRANSFORM = 'transform';
+}
+
+/**
+ * An object that handles the display of a text and/or an icon, linked to a 3D
+ * position. The content of the `Label` is managed through a DOM object, in a
+ * `<div>` handled by the `Label2DRenderer`.
+ *
+ * @property {boolean} isLabel - Used to checkout whether this object is a
+ * Label. Default is true. You should not change this, as it is used internally
+ * for optimisation.
+ * @property {Element} content - The DOM object that contains the content of the
+ * label. The style and the position are applied on this object.
+ * @property {THREE.Vector3} position - The position in the 3D world of the
+ * label.
+ * @property {Coordinates} coordinates - The coordinates of the label.
+ */
+class Label extends THREE.Object3D {
+    /**
+     * @param {Element|string} content - The content; can be a
+     * string, with or without HTML tags in it, or it can be an Element.
+     * @param {Coordinates} coordinates - The world coordinates, where to place
+     * the Label.
+     * @param {Object} options - Some options to apply to the label.
+     * @param {string} options.crs - The CRS, used to place the label on the surface of
+     * the view. Usually it is the CRS of the view.
+     * @param {Style} style - The style to apply to the content. Once the style
+     * is applied, it cannot be changed directly. However, if it really needed,
+     * it can be accessed through `label.content.style`, but it is highly
+     * discouraged to do so.
+     */
+    constructor(content, coordinates, options = {}, style) {
+        if (content == undefined || coordinates == undefined) {
+            throw new Error('content and coordinates are mandatory to add a Label');
+        }
+
+        super();
+
+        let _visible = this.visible;
+        // can't do an ES6 setter/getter here
+        Object.defineProperty(this, 'visible', {
+            set(v) {
+                if (v != _visible) { // avoid changing the style
+                    _visible = v;
+                    this.content.style.display = v ? 'block' : 'none';
+                    // TODO: add smooth transition for fade in/out
+                }
+            },
+            get() {
+                return _visible;
+            },
+        });
+
+        this.isLabel = true;
+        this.coordinates = coordinates;
+
+        if (options.translation && options.scale) {
```

Comment:
Could you move the transformation out of the constructor?
You could add a method in `Feature` to get the good coordinates.

Reply:
Ok, will do